happy-birthday-to-you.js

```diff
@@ -0,0 +1,153 @@
+/* 
+
+------ STEPS ------
+1.  Ensure that name is in correct format (only strings, also account for 
+	special characters).
+
+2. Once correct name format is inputted, compare it to object key/value pairs.
+
+2a. Play a birthday song if dates match. otherwise, play the Rick Roll 
+
+*/
+
+const birthdayObj = {
+
+	'Abdullah Sayeem': 513,
+	'Andre Chow': 103,
+	'Gurdeep Singh': 327,
+	'Jimmy Feng': 711,
+	'John Chen': 1202,
+	'Kevin Chen': 627,
+	'Kevin Cheung':	521,
+	'Mohammed Rahman': 305,
+	'Robert Tan': 523,
+	'Roland Zhou': 903,
+	'Vainius Glinskis': 805,
+	'Zarif Rahat': 1107,
+
+};
+
+function validateInputName() {
+	let text = nameInput.value.toLowerCase();
+	// any combination of letters separated by a whitespace
+	let nameFormat = /^[a-zA-Z]+\s+[a-zA-Z]+$/; 
+
+	return nameFormat.test(text);
+}
+
+function confirmOnList() {
+	let text = nameInput.value.toLowerCase();
+
+	for (let name in birthdayObj) {
+		name = name.toLowerCase();
+
+		if (text === name) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+function getCurrentDate() {
+	const today = new Date();
+	const dd = today.getDate();
+	const mm = today.getMonth() + 1; // January is 0!
+	const yyyy = today.getFullYear();
+
+	return `${mm}${dd}`; 
+}
+
+const nameInputContainer = document.querySelector('.name-input-container');
+const nameInput = document.getElementById('name-input');
+const mainContent = document.querySelector('.main-content');
+const goButton = document.getElementById('go-button');
+
+goButton.addEventListener('click', function(event) {
+
+	// remove last child of both `<div>`s every time go button is clicked*
+	if (nameInputContainer.lastChild.nodeName === 'H6') {
+		nameInputContainer.lastChild.remove();
+	}
+
+	if (nameInputContainer.lastChild.nodeName === 'H1') {
+		mainContent.lastChild.remove();
+	}
+
+	// STEP 1 
+	// Ensure that name is in correct format (only strings)
+
+	// variables for validation
+	let isValidatedOrNot = validateInputName();
+	const heading6 = document.createElement('h6');
+	heading6.style.margin = '5px';
+
+	// variables for confirmation
+	let text = nameInput.value.toLowerCase();
+	let onListOrNot = confirmOnList();
+	let todaysDate = getCurrentDate();
+	const heading1 = document.createElement('h1');
+	const birthdayVideo = document.createElement('iframe');
+	
+	// CASE 1: User inputs empty string
+	if (text === '') {
+		heading6.innerHTML = `Please enter a name.`;
+		nameInputContainer.appendChild(heading6);
+		
+	// CASE 2: User inputs a non-empty string but in the incorrect format
+	} else if (isValidatedOrNot === false) {
+		heading6.innerHTML = `Please enter your name in the correct format.`;
+		nameInputContainer.appendChild(heading6);
+
+	// STEP 2 
+	// Once correct name format is inputted, compare it to object key/value pairs. 
+
+	// CASE 1: User is a member
+	} else if (onListOrNot === true) {
+		for (let name in birthdayObj) {
```

Comment:
ya not sure why we need to iterate again if we already know whether or not they are on list from the `confirmOnList` function

Reply:
I needed a way to access the value of each key in the object to do the comparison:  `(birthdayObj[name] === getCurrentDate()`

Reply:
I see! In the case, you may not even need `confirmOnList` since all it does is return `birthdayObj[name]`. You can just save that to a variable and then perform the date check off that!